### ISSUE TITLE
recall is high and precision is very small

### ISSUE THREAD

**Issue by hust-kevin** [eriklindernoren/PyTorch-YOLOv3#106]

**Comment by MrWhiteHomeman** [comment]
I met the same problem, did  you solve it ?

**Comment by MrWhiteHomeman** [comment]
I met the same problem, did you solve it ?@hust-kevin

**Comment by hajlyx** [comment]
I think that loss_conf has something wrong, I trained a model using pjreddie's darknet code, everything goes well, and I loaded this model to continue training (using this repo), the terminal showed that  loss_conf is about 31 when precision and recall is about 0.7. After training for a couple of batches the loss_conf decreased to 0.8 but the precision also decreased to 0.01. When I tested the model, it came out 27200 detections! Obviously 99% of them were false detection. And the objectness confs of these detections were about 0.999.

**Comment by fengxinjie** [comment]
> I think that loss_conf has something wrong, I trained a model using pjreddie's darknet code, everything goes well, and I loaded this model to continue training (using this repo), the terminal showed that loss_conf is about 31 when precision and recall is about 0.7. After training for a couple of batches the loss_conf decreased to 0.8 but the precision also decreased to 0.01. When I tested the model, it came out 27200 detections! Obviously 99% of them were false detection. And the objectness confs of these detections were about 0.999.

the loss_cls is not right, you must change         pred_cls = torch.sigmoid(prediction[..., 5:]) # Cls pred.
to                    pred_cls = prediction[..., 5:] # Cls pred.

sigmoid is not need for nn.CrossEntropyLoss()

**Comment by M0reDr1nk** [comment]
prediction[...,5:] in code are "the prob of each cls".
but low precision is caused by high fp.
And maybe high fp is caused by wrong conf.
so I think only changing this maybe not work...

**Comment by jw-pyo** [comment]
Hi @hajlyx , I got the same problem in training my own dataset. Is there any progress after then?

**Comment by fengxinjie** [comment]
> Hi @hajlyx , I got the same problem in training my own dataset. Is there any progress after then?

I rewrite yolov3 in pytorch, and I recommend nn.BCEWithLogitsLoss as the classifier loss, it works well

**Comment by jw-pyo** [comment]
> > Hi @hajlyx , I got the same problem in training my own dataset. Is there any progress after then?
> 
> I rewrite yolov3 in pytorch, and I recommend nn.BCEWithLogitsLoss as the classifier loss, it works well

Thanks for your answer @fengxinjie ! But I solved this problem by adjusting the learning rate smaller than original(0.001 -> 0.0002). And, it's a bit silly mistake, I put the annotation files at wrong directory. Hope other people will not make mistake as I've done.

**Comment by fourth-archive** [comment]
I would try this repo, P and R start high and go higher. You can plot your results as well.
https://github.com/ultralytics/yolov3

`from utils import utils; utils.plot_results()`
![Alt](https://user-images.githubusercontent.com/26833433/53494085-3251aa00-3a9d-11e9-8af7-8c08cf40d70b.png "train.py results")
